internal/store/store.go

```diff
@@ -31,11 +31,29 @@ func DeleteRoute(keys ...string) {
 	}
 }
 
-func GetRoute(key string) string {
+func GetRouteOrigin(key string) string {
 	if r, ok := routesMap[key]; ok {
 		return fmt.Sprintf("%s:%s", r.service, r.port)
 	}
 
 	log.Error().Msgf("Service %s not found in routes map", key)
 	return ""
 }
+
+func GetRouteLabel(key string) string {
+	if r, ok := routesMap[key]; ok {
+		return r.label
+	}
+
+	log.Error().Msgf("Service %s not found in routes map", key)
+	return ""
+}
+
+func GetRouteNamespace(key string) string {
+	if r, ok := routesMap[key]; ok {
+		return r.namespace
```

Comment:
yeah see comment regarding read/write locks, you would have 2-3x reads when u could lower it to 1x read an then process the object as is ( comment above regarding double read )